script.js

```diff
@@ -1 +1,44 @@
-// this is script file
+// all the dom elements are selected
+const addTodoInputElement = document.getElementById("addTodoInputField");
+const addTodoButton = document.getElementById("addTodoButton");
+const allTodoContainer = document.getElementById("todosContainer");
+
+// global variables
+let allTodosArray = [];
+
+// input data sanitization by using removing any tags
+const dataSanitization = (value) => {
+    return value.replace(/(<([^>]+)>)/gi, "");
+};
+
+// get the todo title from the input element
+const addTodoHandler = () => {
+    let todoTitle = addTodoInputElement.value;
+    if (!todoTitle.length) {
+        alert("Please enter a valid text");
+        return;
+    }
+
+    allTodosArray.unshift({
+        id: new Date().getTime(),
+        title: dataSanitization(todoTitle),
+    });
+
+    renderTodoArray(allTodosArray);
+};
+
+// render functions to update the dom
+const renderTodoArray = (toBeRenderedArray) => {
+    allTodoContainer.innerHTML = "";
+
+    const finalRenderedTodos = toBeRenderedArray
+        .map((todo) => {
+            return `<li>${todo.title}</li>`;
+        })
+        .join("");
+
+    allTodoContainer.innerHTML = finalRenderedTodos;
+};
```

Comment:
```suggestion
const createTodoElement(todo){
    const $todo = document.createElement("li")
    $todo.innerHTML = `<p>${todo.title}</p>`
    
    return $todo
}

const renderTodos = (todos) => {
    $todos.innerHTML = "";
        
    todos.forEach(todo => {
        $todos.appendChild(createTodoElement(todo))
    })
};
```

Reply:
Bhai do I have to use for each? why not map?By using map I can avoid the appending problems.

Reply:
Nice catch. I've updated my suggestion. Thanks

Reply:
But in general you have to create a todo item for each object in `todos` because each of them will have different eventListeners down the line.

Reply:
okay bhai.